src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -17903,5 +17908,838 @@ public class XAttribute : System.Attribute
+            //   not containing element conversions in the error case.
+            CreateCompilation(src).VerifyEmitDiagnostics(
+                // (2,12): error CS0037: Cannot convert null to 'int' because it is a non-nullable value type
+                // int[] x = [null];
+                Diagnostic(ErrorCode.ERR_ValueCantBeNull, "null").WithArguments("int").WithLocation(2, 12),
+                // (2,12): warning CS8619: Nullability of reference types in value of type '<null>' doesn't match target type 'int'.
+                // int[] x = [null];
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "null").WithArguments("<null>", "int").WithLocation(2, 12),
+                // (3,9): error CS0037: Cannot convert null to 'int' because it is a non-nullable value type
+                // int y = null;
+                Diagnostic(ErrorCode.ERR_ValueCantBeNull, "null").WithArguments("int").WithLocation(3, 9)
+                );
+        }
+
+        [Fact]
+        public void ElementNullability_CheckExpressions()
+        {
+            string src = """
+                #nullable enable
+                object? o = null;
+                string[] x = [o.ToString()];
+                """;
+
+            CreateCompilation(src).VerifyEmitDiagnostics(
+                // (3,15): warning CS8602: Dereference of a possibly null reference.
+                // string[] x = [o.ToString()];
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "o").WithLocation(3, 15)
+                );
+        }
+
+        [Fact]
+        public void SpreadNullability()
+        {
+            string src = """
+                #nullable enable
+                string[]? x1 = null;
+                string[] x2 = new string[0];
+
+                string[] y1 = [.. x1];
+                string[] y2 = [.. x2];
+                """;
+
+            // https://github.com/dotnet/roslyn/issues/68786: We should check the spreads
+            CreateCompilation(src).VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void SpreadNullability_CheckExpressions()
+        {
+            string src = """
+                #nullable enable
+                string[] y1 = [.. M(null)];
+                string[] y2 = [.. M(new object())];
+
+                string[] M(object o) => throw null!;
+                """;
+
+            CreateCompilation(src).VerifyEmitDiagnostics(
+                // (2,21): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                // string[] y1 = [.. M(null)];
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(2, 21)
+                );
+        }
+
+        [Fact]
+        public void SpreadNullability_SplitExpression()
+        {
+            // https://github.com/dotnet/roslyn/issues/68786: We should check the spreads without asserting in DebugVerifier
+#if RELEASE
+            string src = """
+                #nullable enable
```

Comment:
Formatting warning is being reported here. I honestly don't know what's wrong. Maybe a bug due to presence of the if-directive above? Might want to use compilation options to enable nullability for this test, or use attributes to only run the test in release mode.

I think @Youssef1313 may have reported something like this recently.

Reply:
@RikkiGibson Yup, that's the same root cause as https://github.com/dotnet/csharplang/discussions/7520.

For "Debug" where the `#if RELEASE` is false, the compiler will parse everything in the `#if RELEASE` block as disabled trivia, and it will parse `#nullable enable` as NullableDirectiveTrivia even if the human thinking is that it's part of a string literal and not a nullable directive trivia.